components/uniset/src/enum_prop_mapping.rs

```diff
@@ -0,0 +1,876 @@
+        }
+        Some(EnumeratedProperty::DecompositionType) => {
+            get_decomposition_type_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::EastAsianWidth) => {
+            get_east_asian_width_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::GeneralCategory) => {
+            get_general_category_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::GraphemeClusterBreak) => {
+            get_grapheme_cluster_break_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::HangulSyllableType) => {
+            get_hangul_syllable_type_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::IndicPositionalCategory) => {
+            get_indic_positional_category_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::IndicSyllabicCategory) => {
+            get_indic_syllabic_category_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::JoiningGroup) => {
+            get_joining_group_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::JoiningType) => get_joining_type_enum(prop_val).map(|x| x as i32),
+        Some(EnumeratedProperty::LineBreak) => get_line_break_enum(prop_val).map(|x| x as i32),
+        Some(EnumeratedProperty::LeadCanonicalCombiningClass) => {
+            get_lead_canonical_combining_class_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::NFCQuickCheck) => {
+            get_nfc_quick_check_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::NFDQuickCheck) => {
+            get_nfd_quick_check_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::NFKCQuickCheck) => {
+            get_nfkc_quick_check_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::NFKDQuickCheck) => {
+            get_nfkd_quick_check_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::NumericType) => get_numeric_type_enum(prop_val).map(|x| x as i32),
+        Some(EnumeratedProperty::SentenceBreak) => {
+            get_sentence_break_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::TrailCanonicalCombiningClass) => {
+            get_trail_canonical_combining_class_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::VerticalOrientation) => {
+            get_vertical_orientation_enum(prop_val).map(|x| x as i32)
+        }
+        Some(EnumeratedProperty::WordBreak) => get_word_break_enum(prop_val).map(|x| x as i32),
+        _ => None,
+    };
+    let name_enum_i32_opt = name_enum_opt.map(|x| x as i32);
+    match (name_enum_i32_opt, val_enum_i32_opt) {
+        (Some(name_i32), Some(val_i32)) => Some((name_i32, val_i32)),
+        _ => None,
+    }
+}
+
+pub fn get_prop_name_identifier(prop_name: &str, prop_val: &str) -> Option<String> {
```

Comment:
This can return a TinyString16

Reply:
Done. At least here, I changed this function to return `Option<TinyStr16>` instead of `Option<String>`.

I thought the long-term intention was that we could, in the future, plumb the `TinyStr16` returned here to be re-used as the identifier string for the `UnicodeProperty` struct returned by the data provider. I at least tried following through that line of thinking, but ran into the problem that if I change the id field of `UnicodeProperty` from `String` to `TinyStr16`, the error is that TinyStr is not currently serde serializable:

```
the trait bound `tinystr::TinyStr16: provider::_::_serde::Serialize` is not satisfied

the trait `provider::_::_serde::Serialize` is not implemented for `tinystr::TinyStr16`

note: required because of the requirements on the impl of `provider::_::_serde::Serialize` for `std::borrow::Cow<'_, tinystr::TinyStr16>`
note: required by `provider::_::_serde::ser::SerializeStruct::serialize_field`
```

Reply:
You need to enable the "serde" feature on the "tinystr" dependency for this to work.  But let's resolve that later. #550